File: src/Tutor.Models/Student.cs
namespace Tutor.Models;

public class Student
{
    public Guid Id { get; set; }
    public string Name { get; set; }
    public string? Description { get; set; }
    public string? avatarUrl { get; set; }
    public int DefaultPrice { get; set; }
    public List<Lesson> Lessons { get; set; } = [];
}


public class StudentSiteDto
{
    public Guid Id { get; set; }
    public string Name { get; set; }
    public string? Description { get; set; }
    public string? avatarUrl { get; set; }
    public int DefaultPrice { get; set; }
}

public static class StudentExtensions
{
    public static StudentSiteDto ToSiteDto(this Student student)
    {
        return new StudentSiteDto
        {
            Id = student.Id,
            Name = student.Name,
            Description = student.Description,
            avatarUrl = student.avatarUrl,
            DefaultPrice = student.DefaultPrice,
        };
    }
}

File: src/Chroniq.WebClient/src/features/main/types/student.ts
import { uuid } from "../lib"

export type Student = {
  id: string
  name: string
  defaultPrice: number
  avatarUrl: string
  description: string
}

export function createStudent(student?: Partial<Student>): Student {
  return { id: uuid(), name: "", avatarUrl: "", description: "", defaultPrice: 1500, ...student }
}

export function isStudentValid(student: Student) {
  return student.name.length > 0 && student.defaultPrice > 0
}
